src/SiteInformation/FarmDates/EditDatesModal.js

```diff
@@ -0,0 +1,165 @@
+import { callAzureFunction } from '../../utils/SharedFunctions';
+import { useAuth0 } from '../../Auth/react-auth0-spa';
+import PropTypes from 'prop-types';
+
+const EditDatesModal = ({ setSnackbarDataGlobal }) => {
+    const editDatesModalData = useSelector((state) => state.farmDatesData.editDatesModalData);
+    const editDatesModalOpen = useSelector((state) => state.farmDatesData.editDatesModalOpen);
+    const [datesDict, setDatesDict] = useState({
+        coverCropPlanting : 'undefined',
+        coverCropTermination : 'undefined',
+        cashPlanting : 'undefined',
+    });
+    const [farmCode, setFarmCode] = useState();
+    const {getTokenSilently} = useAuth0();
+    const dispatch = useDispatch();
+
+    //set all values when first loaded
+    useEffect(() => {
+        setFarmCode(editDatesModalData.code);
+        setDatesDict({
+            coverCropPlanting: editDatesModalData.cover_planting,
+            coverCropTermination: editDatesModalData.cover_termination,
+            cashPlanting: editDatesModalData.cash_planting,
+        });
+    }, [editDatesModalData]);
+
+    //updates the dictionary with the passed data based on which column is selected
+    //0=coverCropPlanting, 1=coverCropTermination, 2=cashPlanting
+    const updateDate = (col, data) => {
+        if (col == 0) {
+            setDatesDict({
+                ...datesDict,
+                coverCropPlanting: data,
+            });
+        } else if (col == 1){
+            setDatesDict({
+                ...datesDict,
+                coverCropTermination: data,
+            });
+        } else if (col == 2){
+            setDatesDict({
+                ...datesDict,
+                cashPlanting: data,
+            });
+        } else {
+            console.log("Invalid column selected");
+        }
+    };
+
+    //handles closing of the modal
+    const handleEditDatesModalClose = () => {
+        dispatch(setEditDatesModalOpen(false));
+    };
+
+    //handles submit
+    const handleEditDatesModalSubmit = () => {
+        //set data to be submitted to API
+        const data = {
+            "values": {
+                "cc_planting_date": datesDict.coverCropPlanting,
+                "cc_termination_date": datesDict.coverCropTermination,
+                "cash_crop_planting_date": datesDict.cashPlanting,
+            },
+            "conditions": {
+                "code": farmCode,
+            }
+        };
+        callAzureFunction( data, 'crowndb/farm_history', 'POST', getTokenSilently).then((res) => {
+            if (res.response && res.response.status === 201) {
+              dispatch(setEditDatesModalData({
+                    ...editDatesModalData,
+                    cover_planting: datesDict.coverCropPlanting,
+                    cover_termination: datesDict.coverCropTermination,
+                    cash_planting: datesDict.cashPlanting,
+              }));
+              dispatch(setFarmDatesReloadNeeded(true));
```

Comment:
this should cause a refetch so you should be able to remove the above

Reply:
also you want to do `setFarmDatesReloadNeeded(!farmDatesReloadNeeded)` instead of `setFarmDatesReloadNeeded(true)` because if you do it twice the value of `farmDatesReloadNeeded` will already be `true` and it won't enter the `useEffect` that has `farmDatesReloadNeeded` as a dependency because it wouldn't change